stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -352,19 +317,14 @@ function mul!(C::AbstractMatrix, Da::Diagonal, Db::Diagonal, alpha::Number, beta
     return C
 end
 
-(/)(Da::Diagonal, Db::Diagonal) = Diagonal(Da.diag ./ Db.diag)
-
-ldiv!(x::AbstractArray, A::Diagonal, b::AbstractArray) = (x .= A.diag .\ b)
-
-function ldiv!(D::Diagonal, A::Union{LowerTriangular,UpperTriangular})
-    broadcast!(\, parent(A), D.diag, parent(A))
-    A
-end
+(/)(A::AbstractVecOrMat, D::Diagonal) =
+    rdiv!(copy_similar(A, promote_op(/, eltype(A), eltype(D))), D)
+(/)(Da::Diagonal, Db::Diagonal) = Diagonal(Db \ Da.diag)
```

Comment:
This and L345 was changed to make it call `ldiv!` rather than broadcast directly to reuse the input checks.

I just realize that the following is correct only for `Eltype <: Number`
```julia
A / B = A * B⁻¹ = (B⁻¹ᵀ * Aᵀ)ᵀ = B⁻¹ᵀ * Aᵀ = B⁻¹ * A = B \ A
```
since `/` perform copy while `\` not, shall we replace this with
```julia
(/)(Da::Diagonal{<:Number}, Db::Diagonal{<:Number}) = Diagonal(Db \ Da.diag)
(/)(Da::Diagonal, Db::Diagonal) = Diagonal(Da.diag ./ Db.diag)
```

Reply:
Excellent point about the `SingularException` check! But isn't this just `Diagonal(Da.diag / Db)`? I'm not sure the code bloat (with `_rdiv!`) is worth it here. Line 345 is able to avoid the first copy-pass "for free" (due to the 3-arg `ldiv!`).

Reply:
In fact I tried that, and then notice that `Da.diag/Db` only works for 1*1 Db.

Reply:
🤦 True!

Reply:
> I just realize that the following is correct only for `Eltype <: Number`

This is not even correct for that case! The third step is wrong, you can't just omit the outer transpose.

Reply:
That should be ok?
Since `(B⁻¹ᵀ * Aᵀ)` isa `Diagnal` and we have `transpose(D::Diagonal{<:Number}) = D`